pandas/core/tools/datetimes.py

```diff
@@ -229,13 +229,12 @@ def _return_parsed_timezone_results(result, timezones, tz, name):
     -------
     tz_result : Index-like of parsed dates with timezone
     """
-    if tz is not None:
-        raise ValueError(
-            "Cannot pass a tz argument when parsing strings with timezone information."
-        )
     tz_results = np.array(
         [Timestamp(res).tz_localize(zone) for res, zone in zip(result, timezones)]
     )
+    if tz is not None:
+        # GH 32792
+        tz_results = np.array([tz_result.astimezone(tz) for tz_result in tz_results])
```

Comment:
Could you use `tz_convert` here instead of `astimezone`?

Reply:
@mroeschke sorry but i'm not aware of any ```tz_convert``` which is for a pd.Index which object dtype

Reply:
`tz_convert` is a `Timestamp` method.